tests/integrationtest/r/expression/issues.result

```diff
@@ -2512,6 +2512,66 @@ select * from t where t.b < t.c;
 b	c
 z	26.18040000000000000000
 drop table if exists t1;
+CREATE TABLE t1 (c1 INT);
+INSERT INTO t1 VALUES (1), (null);
+SELECT IFNULL(c1, 0.0) from t1;
+IFNULL(c1, 0.0)
+1.0
+0.0
+SELECT if(c1 is not null, c1, 0.0) from t1;
+if(c1 is not null, c1, 0.0)
+1.0
+0.0
+SELECT case when c1 is not null then c1 else 0.0 end from t1;
+case when c1 is not null then c1 else 0.0 end
+1.0
+0.0
+drop table if exists t1;
+CREATE TABLE t1 (v1 datetime(3), v2 datetime(4));
+INSERT INTO t1 VALUES ('2020-01-01 11:11:11.123', null), (null, '2020-01-01 11:11:11.1234');
+SELECT IFNULL(v1, v2) from t1;
+IFNULL(v1, v2)
+2020-01-01 11:11:11.1230
+2020-01-01 11:11:11.1234
+SELECT if(v1 is not null, v1, v2) from t1;
+if(v1 is not null, v1, v2)
+2020-01-01 11:11:11.1230
+2020-01-01 11:11:11.1234
+SELECT case when v1 is not null then v1 else v2 end from t1;
+case when v1 is not null then v1 else v2 end
+2020-01-01 11:11:11.1230
+2020-01-01 11:11:11.1234
+drop table if exists t1;
+CREATE TABLE t1 (v1 timestamp(3), v2 timestamp(4));
+INSERT INTO t1 VALUES ('2020-01-01 11:11:11.123', null), (null, '2020-01-01 11:11:11.1234');
+SELECT IFNULL(v1, v2) from t1;
+IFNULL(v1, v2)
+2020-01-01 11:11:11.1230
+2020-01-01 11:11:11.1234
+SELECT if(v1 is not null, v1, v2) from t1;
+if(v1 is not null, v1, v2)
+2020-01-01 11:11:11.1230
+2020-01-01 11:11:11.1234
+SELECT case when v1 is not null then v1 else v2 end from t1;
+case when v1 is not null then v1 else v2 end
+2020-01-01 11:11:11.1230
+2020-01-01 11:11:11.1234
+drop table if exists t1;
+CREATE TABLE t1 (var_fld int(11) DEFAULT '0' );
+INSERT INTO t1 VALUES (4500), (6000);
+SELECT IF ( var_fld  > 5000, ( 00001 / var_fld ) , 5000 ) if_var FROM t1;
+if_var
+5000.0000
+0.0002
+SELECT CASE WHEN var_fld  > 5000 THEN ( 1 / var_fld ) ELSE 5000 END case_var FROM t1;
+case_var
+5000.0000
+0.0002
+SELECT CASE var_fld  > 5000 WHEN TRUE THEN ( 1 / var_fld ) ELSE 5000 END case_var FROM t1;
+case_var
+5000.0000
```

Comment:
In the original code it would be 99.9999 because 5000 would be cast to decimal(6, 4)

Reply:
Does it mean it will be a bug in the original code?

Reply:
yes, fixed in https://github.com/pingcap/tidb/pull/46786/files#diff-b881614a44394d679db8bf45934c5d7b00b1600bb70da6c9a1b71b1e1e7c0173R232-R236